opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -232,11 +285,248 @@ message MetricDescriptor {
+    //
+    // When SNAPSHOT is not set, it implies the event originated from
+    // the application calling the SDK with a measurement with a
+    // context.  These events are CONTINUOUS measurements, in that
+    // case.
+    SNAPSHOT = 0x10;
+  }
+
+  // Kind explains how the DataPoint was produced (called
+  // "Structure"), how the point was aggregated with resepect to time
+  // (called "Temporality"), whether it was computed by a snapshot,
+  // and (when the Structure is ADDING) whether the sum is not
+  // monotonic.
+  //
+  // Kind names are generated from valid combinations of KindElement
+  // by joining the effective KindElements using underscores.  There
+  // are:
+  //
+  // - 3 possibilities for Temporality
+  // - 3 possibilities for Structure:
+  // - 2 possibilities of being a Snapshot or not (i.e., Continuous).
+  //
+  // Excluding the snapshot instantaneous combinations, this makes
+  // 15 valid values.
+  //
+  // The cases are detailed below in terms of:
+  //
+  // - How StartTimeUnixNano and TimeUnixNano are interpreted?
+  // - Which OpenTelemetry is likely to produce instrument?
+  // - Under which typical aggregations?
+  // - Producing which kind of data point?
+  //
+  // Certain general statements can be made.
+  //
+  // About the three Structure kinds:
+  //
+  // 1. ADDING_MONOTONIC kinds associate with Counter and SumObserver
+  // 2. ADDING kinds associate with UpDownCounter and UpDownSumObserver
+  // 3. GROUPING kinds associate with ValueRecorder and ValueObserver instruments
+  //
+  // About SNAPSHOT kinds:
+  // 
+  // 1. SNAPSHOT kinds associate with Observer instruments
+  // 2. INSTANTANEOUS temporality does not combine with SNAPSHOT kinds.
+  //
+  // About ADDING|INSTANTANEOUS vs ADDING|DELTA:
+  //
+  // 1. An ADDING|INSTANTANEOUS event is stateless, has no StartTimeUnixNano.
+  // 2. An ADDING|DELTA event remembers its start time, sets StartTimeUnixNano.
+  // 3. Same consideration for ADDING_MONOTONIC instantaneous vs delta.
+  //
+  // About SCALAR_INT64 and SCALAR_DOUBLE value type (TODO(#159) and
+  // equally the RAW value types):
+  //
+  // 1. SCALAR value type interpretation depends on both temporality
+  //    and structure.
+  // 2. For GROUPING kinds:
+  //    - For INSTANTANEOUS temporality: An individual measurement
+  //    - For DELTA or CUMULATIVE temporality: The last measured individual value
+  //      in the collection interval.
+  // 3. For ADDING and ADDING_MONOONIC kinds:
+  //    - For INSTANTANEOUS temporality: An instantaneously measured change in a sum
+  //    - For DELTA temporality: An aggregated change in a sum
+  //    - For CUMULATIVE temporality: An aggregated sum.
+  // 4. INSTANTANEOUS kinds are always scalar values (TODO(#159) or raw).
+  //
+  // About CUMULATIVE kinds:
+  //
+  // 1. CUMULATIVE GROUPING kinds typically use Histogram data points
+  // 2. CUMULATIVE is rarely used with Summary data points
+  enum Kind {
```

Comment:
Spent time during the weekend trying this new proposal, and found a lot of things that don't make a lot of sense.

See more details here and a proposed different approach of thinking about the data model:
https://docs.google.com/document/d/1UBpke_e_HWl7UV3piNIMAZYw8eNjfGjucdwwsEOqUI4/edit#

Reply:
Please don't hesitate to comment in the doc or here. Will try to answer all the comments :)